.github/copilot-instructions.md

```diff
@@ -0,0 +1,212 @@
+- **gradle.properties**: JVM memory settings and Kotlin MPP configuration
+- **settings.gradle.kts**: Project structure and repository configuration
+
+### Source Set Structure (Per Module)
+```
+src/
+├── commonMain/           # Shared APIs across all platforms
+├── composeUiMain/        # Compose UI targets (JVM, Android, iOS, Wasm)
+├── composeUiExceptAndroidMain/  # Compose UI excluding Android
+├── androidMain/          # Android-specific implementations
+├── jsMain/               # JS DOM implementations (Kobweb/HTML)
+├── jvmMain/              # Desktop JVM implementations
+└── iosMain/              # iOS-specific implementations
+```
+
+### CI/CD Workflows
+
+#### `.github/workflows/ci.yml`
+- **Trigger**: Push to any branch, manual dispatch
+- **Platforms**: ubuntu-latest, macos-latest, windows-latest  
+- **JDK**: 17-temurin
+- **Commands**: Uses `huanshankeji/.github/actions/gradle-test-and-check@v0.2.0`
+
+#### `.github/workflows/demo-gh-pages.yml`
+- **Trigger**: Push/PR to `release` branch, manual dispatch
+- **Platform**: ubuntu-latest
+- **Purpose**: Builds and deploys demo to GitHub Pages
+- **Build Command**: `./gradlew :compose-multiplatform-html-unified-demo:sideBySideBrowserDistribution`
+- **Artifact Path**: `demo/build/dist/sideBySide/productionExecutable/`
+
+### Dependencies and External Integration
+
+#### Required External Repositories
+```kotlin
+repositories {
+    mavenCentral()
+    google()  // For Android components
+}
+```
+
+**Note**: All dependencies are available from standard repositories. If working with snapshot dependencies of other `com.huanshankeji` libraries, additional setup may be required as per organization instructions.
+
+#### Key Dependencies
+- **Kotlin**: 2.2.20 with Compose Compiler
+- **Compose Multiplatform**: 1.9.0
+- **Android Gradle Plugin**: 8.11.2  
+- **Kobweb**: 0.23.3 (now on Maven Central)
+- **Compose HTML Material**: 0.4.1-SNAPSHOT
+- **Binary Compatibility Validator**: 0.18.1 (enabled)
+
+### Platform-Specific Implementation Patterns
+
+#### Compose UI Platforms (Android, JVM, iOS, Wasm)
+- Delegate directly to official `androidx.compose` APIs
+- Located in `composeUiMain` source sets
+
+#### JS DOM Platform  
+- Built on **Kobweb Silk** (provides Modifier system and layouts)
+- Uses **Material Web Components** via Compose HTML Material
+- Located in `jsMain` source sets
+- Requires CSS imports for Material Icons
+
+### Development Patterns and Conventions
+
+#### API Structure
+- **Common APIs**: Defined in `commonMain`, expect/actual pattern for platform differences
+- **Ext Components**: Components in `ext` packages don't follow androidx.compose APIs exactly but provide more idiomatic platform-specific wrappers
+- **Parameter Naming**: Parameters with "JsDom" or "ComposeUi" suffixes are platform-specific
+
+#### Code Quality and Validation
+- **Binary Compatibility**: Enforced via kotlinx.validation plugin
```

Comment:
The plugin's full ID is "org.jetbrains.kotlinx.binary-compatibility-validator" and the repo is <https://github.com/Kotlin/binary-compatibility-validator>. Where did you get the name "kotlinx.validation"?